module/VuFindTheme/tests/unit-tests/src/VuFindTest/ThemeInfoTest.php

```diff
@@ -339,15 +300,67 @@ public function testGetMergedConfigNoKey()
         );
     }
 
+    /**
+     * Stress-test our merging algorithm
+     *
+     * @return void
+     */
+    public function testMergeWithoutOverrideEdgeCases()
+    {
+        $ti = $this->getThemeInfo();
+
+        // string
+        $merged = $ti->mergeWithoutOverride('original', 'override');
```

Comment:
If you want to directly test mergeWithoutOverride without having to actually make the method public, you can `use \VuFindTest\Feature\ReflectionTrait;` at the top of the test class, and then use (for example) `$this->callMethod($ti, 'mergeWithoutOverride', ['original', 'override'])`.

Reply:
Refactoring this part of the test to use a dataProvider might also make things a bit cleaner, since then you could have a series of named test cases in the data provider, and you'd only have to rely on the reflection call in one spot in the code.